group1/comm/static/comm_tool.js

```diff
@@ -825,3 +823,230 @@ function downloadFileCallback(data) {
+            {"optional": [{"DtlsSrtpKeyAgreement": true}]} /* this will no longer be needed by chrome
+                                                                        * eventually (supposedly), but is necessary
+                                                                        * for now to get firefox to talk to chrome */
+        );
+
+        peers[peer_id]=peer_connection;
+
+        peer_connection.onicecandidate = function(event) {
+            if (event.candidate) {
+                video_socket.emit('relayICECandidate', {
+                    'peer_id': peer_id,
+                    'ice_candidate': {
+                        'sdpMLineIndex': event.candidate.sdpMLineIndex,
+                        'candidate': event.candidate.candidate
+                    }
+                });
+            }
+        }
+        peer_connection.onaddstream = function(event) {
+            console.log("onAddStream", event);
+            var remote_media = USE_VIDEO ? $("<video>") : $("<audio>");
+            remote_media.attr("autoplay", "autoplay");
+            if (MUTE_AUDIO_BY_DEFAULT) {
+                remote_media.attr("muted", "true");
+            }
+            //remote_media.attr("controls", "");
+            //peer_media_elements[peer_id] = remote_media;
+            //$('body').append(remote_media);
+            //attachMediaStream(remote_media[0], event.stream);
+            var localVideo=document.getElementById("localVideo");
+            localVideo.src=window.URL.createObjectURL(event.stream);
+
+        }
+        peer_connection.addStream(local_media_stream);
+        if (config.should_create_offer) {
+            console.log("Creating RTC offer to ", peer_id);
+            peer_connection.createOffer(
+                function (local_description) {
+                    console.log("Local offer description is: ", local_description);
+                    peer_connection.setLocalDescription(local_description,
+                        function() {
+                            video_socket.emit('relaySessionDescription',
+                                {'peer_id': peer_id, 'session_description': local_description});
+                            console.log("Offer setLocalDescription succeeded");
+                        },
+                        function() { Alert("Offer setLocalDescription failed!"); }
+                    );
+                },
+                function (error) {
+                    console.log("Error sending offer: ", error);
+                });
+        }
+
+    });
+    video_socket.on('sessionDescription', function(config) {
+        console.log('Remote description received: ', config);
+        var peer_id = config.peer_id;
+        var peer = peers[peer_id];
+        var remote_description = config.session_description;
+        console.log(config.session_description);
+
+        var desc = new RTCSessionDescription(remote_description);
+        var stuff = peer.setRemoteDescription(desc,
```

Comment:
Can var stuff be renamed to something more indicative of what it is?  Maybe "remote_description_promise"?

Reply:
adjusted